### PR TITLE
Port System.Threading new docs

### PR DESCRIPTION
**Pull request by carlossanlop**
@dotnet/area-system-threading-tasks @dotnet/area-system-threading-channels @mangod9 @kouvel @directhex @gewarren PTAL

### CODE REVIEW

File: xml/System.Threading.Tasks/Task.xml [hunk 1]
```diff
@@ -6463,10 +6482,20 @@ timeout is greater than
...
       <Docs>
-        <param name="tasks">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="tasks">The tasks to wait on for completion.</param>
+        <summary>Creates a task that will complete when all of the supplied tasks have completed.</summary>
+        <returns>A task that represents the completion of all of the supplied tasks.</returns>
+        <remarks>
+          <para> If any of the supplied tasks completes in a faulted state, the returned task will also complete in a Faulted state, where its exceptions will contain the aggregation of the set of unwrapped exceptions from each of the supplied tasks.
```
Comment (tannergooding) [inline xml/System.Threading.Tasks/Task.xml:6489]: There's a leading whitespace between the para tag and the first word on many of this, is that intentional?

**Review by tannergooding — APPROVED** [review]
Threading.Tasks LGTM

File: xml/System.Threading.Channels/Channel.xml [hunk 2]
```diff
@@ -302,11 +304,12 @@
...
-        <typeparam name="T">To be added.</typeparam>
-        <param name="options">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <typeparam name="T">Specifies the type of data in the channel.</typeparam>
+        <param name="options">Options that guide the behavior of the channel.</param>
+        <summary>Creates an unbounded prioritized channel subject to the provided options.</summary>
+        <returns>The created channel.</returns>
+        <remarks>The supplied <paramref name="options" />' <see cref="P:System.Threading.Channels.UnboundedPrioritizedChannelOptions`1.Comparer" /> is used to determine priority of elements, or <see cref="P:System.Collections.Generic.Comparer`1.Default" /> if the provided comparer is <see langword="null" />.
```
Comment (gewarren) [inline xml/System.Threading.Channels/Channel.xml:311]:
```suggestion
        <remarks>The <see cref="P:System.Threading.Channels.UnboundedPrioritizedChannelOptions`1.Comparer" /> of the supplied <paramref name="options" /> is used to determine priority of elements, or <see cref="P:System.Collections.Generic.Comparer`1.Default" /> if the provided comparer is <see langword="null" />.
```

File: xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml [hunk 3]
```diff
@@ -24,8 +24,9 @@
     </Attribute>
   </Attributes>
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Provide extensions methods for <see cref="T:System.Threading.Tasks.Task" /> operations with <see cref="T:System.TimeProvider" />.</summary>
```
Comment (gewarren) [inline xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml:27]:
```suggestion
    <summary>Provides extensions methods for <see cref="T:System.Threading.Tasks.Task" /> operations with <see cref="T:System.TimeProvider" />.</summary>
```

File: xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml [hunk 4]
```diff
@@ -24,8 +24,9 @@
...
+    <remarks>The Microsoft.Bcl.TimeProvider library interfaces are intended solely for use in building against pre-.NET 8 surface area.
+ If your code is being built against .NET 8 or higher, then this library should not be utilized.</remarks>
```
Comment (gewarren) [inline xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml:29]:
```suggestion
 If your code is being built against .NET 8 or later, then don't use this library.</remarks>
```

File: xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml [hunk 5]
```diff
@@ -48,11 +49,19 @@
...
-        <remarks>To be added.</remarks>
+        <param name="timeProvider">The <see cref="T:System.TimeProvider" /> with which to interpret the <paramref name="delay" />.</param>
+        <param name="delay">The time interval to wait before canceling this <see cref="T:System.Threading.CancellationTokenSource" />.</param>
+        <summary>Initializes a new instance of the <see cref="T:System.Threading.CancellationTokenSource" /> class that will be canceled after the specified <see cref="T:System.TimeSpan" />.</summary>
+        <returns>
+          <see cref="T:System.Threading.CancellationTokenSource" /> that will be canceled after the specified <paramref name="delay" />.</returns>
+        <remarks>
+          <para> The countdown for the delay starts during the call to the constructor. When the delay expires, the constructed <see cref="T:System.Threading.CancellationTokenSource" /> is canceled if it has not been canceled already.
+ </para>
+          <para> If running on .NET versions earlier than .NET 8.0, there is a constraint when invoking <see cref="M:System.Threading.CancellationTokenSource.CancelAfter(System.TimeSpan)" /> on the resultant object.
```
Comment (gewarren) [inline xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml:60]:
```suggestion
          <para> If running on .NET versions earlier than .NET 8, there is a constraint when invoking <see cref="M:System.Threading.CancellationTokenSource.CancelAfter(System.TimeSpan)" /> on the resultant object.
```

File: xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml [hunk 6]
```diff
@@ -48,11 +49,19 @@
...
+ This action will not terminate the initial timer indicated by <paramref name="delay" />. However, this restriction does not apply on .NET 8.0 and later versions.
```
Comment (gewarren) [inline xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml:61]:
```suggestion
 This action will not terminate the initial timer indicated by <paramref name="delay" />. However, this restriction does not apply on .NET 8 and later versions.
```

File: xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml [hunk 7]
```diff
@@ -48,11 +49,19 @@
...
+ </para>
+        </remarks>
+        <exception cref="T:System.ArgumentOutOfRangeException">The <paramref name="delay" /> is negative and not equal to <see cref="F:System.Threading.Timeout.InfiniteTimeSpan" /> or greater than maximum allowed timer duration.</exception>
```
Comment (gewarren) [inline xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml:64]:
```suggestion
        <exception cref="T:System.ArgumentOutOfRangeException">The <paramref name="delay" /> is negative and not equal to <see cref="F:System.Threading.Timeout.InfiniteTimeSpan" />, or is greater than the maximum allowed timer duration.</exception>
```

File: xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml [hunk 8]
```diff
@@ -104,13 +116,17 @@
...
+        <param name="timeout">The timeout after which the <see cref="T:System.Threading.Tasks.Task" /> should be faulted with a <see cref="T:System.TimeoutException" /> if it hasn't otherwise completed.</param>
+        <param name="timeProvider">The <see cref="T:System.TimeProvider" /> with which to interpret <paramref name="timeout" />.</param>
+        <param name="cancellationToken">The <see cref="T:System.Threading.CancellationToken" /> to monitor for a cancellation request.</param>
+        <summary>Gets a <see cref="T:System.Threading.Tasks.Task" /> that will complete when this <see cref="T:System.Threading.Tasks.Task" /> completes, when the specified timeout expires, or when the specified <see cref="T:System.Threading.CancellationToken" /> has cancellation requested.</summary>
+        <returns>The <see cref="T:System.Threading.Tasks.Task" /> representing the asynchronous wait.  It may or may not be the same instance as the current instance.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">The <paramref name="task" /> argument is <see langword="null" />.</exception>
```
Comment (gewarren) [inline xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml:126]:
```suggestion
        <exception cref="T:System.ArgumentNullException">
          <paramref name="task" /> or <paramref name="timeProvider" /> is <see langword="null" />.</exception>
```

File: xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml [hunk 9]
```diff
@@ -104,13 +116,17 @@
...
+        <exception cref="T:System.ArgumentNullException">The <paramref name="timeProvider" /> argument is <see langword="null" />.</exception>
```
Comment (gewarren) [inline xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml:127]:
```suggestion
```

File: xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml [hunk 10]
```diff
@@ -145,13 +161,17 @@
...
+        <param name="timeout">The timeout after which the <see cref="T:System.Threading.Tasks.Task" /> should be faulted with a <see cref="T:System.TimeoutException" /> if it hasn't otherwise completed.</param>
+        <param name="timeProvider">The <see cref="T:System.TimeProvider" /> with which to interpret <paramref name="timeout" />.</param>
+        <param name="cancellationToken">The <see cref="T:System.Threading.CancellationToken" /> to monitor for a cancellation request.</param>
+        <summary>Gets a <see cref="T:System.Threading.Tasks.Task" /> that will complete when this <see cref="T:System.Threading.Tasks.Task" /> completes, when the specified timeout expires, or when the specified <see cref="T:System.Threading.CancellationToken" /> has cancellation requested.</summary>
+        <returns>The <see cref="T:System.Threading.Tasks.Task" /> representing the asynchronous wait.  It may or may not be the same instance as the current instance.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">The <paramref name="task" /> argument is <see langword="null" />.</exception>
```
Comment (gewarren) [inline xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml:171]:
```suggestion
        <exception cref="T:System.ArgumentNullException">
          <paramref name="task" /> or <paramref name="timeProvider" /> is <see langword="null" />.</exception>
```

File: xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml [hunk 11]
```diff
@@ -145,13 +161,17 @@
...
+        <exception cref="T:System.ArgumentNullException">The <paramref name="timeProvider" /> argument is <see langword="null" />.</exception>
```
Comment (gewarren) [inline xml/System.Threading.Tasks/TimeProviderTaskExtensions.xml:172]:
```suggestion
```

File: xml/System.Threading.Tasks/TaskCompletionSource`1.xml [hunk 12]
```diff
@@ -535,9 +535,16 @@
         <Parameter Name="completedTask" Type="System.Threading.Tasks.Task&lt;TResult&gt;" Index="0" FrameworkAlternate="net-9.0" />
       </Parameters>
       <Docs>
-        <param name="completedTask">To be added.</param>
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="completedTask">The completed task whose completion status (including result, exception, or cancellation information) should be copied to the underlying task.</param>
+        <summary>Transition the underlying <see cref="T:System.Threading.Tasks.Task`1" /> into the same completion state as the specified <paramref name="completedTask" />.</summary>
```
Comment (gewarren) [inline xml/System.Threading.Tasks/TaskCompletionSource`1.xml:539]:
```suggestion
        <summary>Transitions the underlying <see cref="T:System.Threading.Tasks.Task`1" /> into the same completion state as the specified <paramref name="completedTask" />.</summary>
```

File: xml/System.Threading.Tasks/TaskCompletionSource.xml [hunk 13]
```diff
@@ -302,9 +302,16 @@ The <xref:System.Threading.Tasks.Task> created by this instance and accessible t
         <Parameter Name="completedTask" Type="System.Threading.Tasks.Task" Index="0" FrameworkAlternate="net-9.0" />
       </Parameters>
       <Docs>
-        <param name="completedTask">To be added.</param>
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="completedTask">The completed task whose completion status (including exception or cancellation information) should be copied to the underlying task.</param>
+        <summary>Transition the underlying <see cref="T:System.Threading.Tasks.Task`1" /> into the same completion state as the specified <paramref name="completedTask" />.</summary>
```
Comment (gewarren) [inline xml/System.Threading.Tasks/TaskCompletionSource.xml:306]:
```suggestion
        <summary>Transitions the underlying <see cref="T:System.Threading.Tasks.Task`1" /> into the same completion state as the specified <paramref name="completedTask" />.</summary>
```

File: xml/System.Threading.Tasks/Task.xml [hunk 14]
```diff
@@ -5436,9 +5442,16 @@ timeout is greater than
...
-        <summary>To be added.</summary>
+        <param name="tasks">An array of <see cref="T:System.Threading.Tasks.Task" /> instances on which to wait.</param>
+        <summary>Waits for all of the provided <see cref="T:System.Threading.Tasks.Task" /> objects to complete execution.</summary>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">The <paramref name="tasks" /> argument contains a <see langword="null" /> element.</exception>
+        <exception cref="T:System.AggregateException">At least one of the <see cref="T:System.Threading.Tasks.Task" /> instances was canceled
+
+-or-
+
+an exception was thrown during
```
Comment (gewarren) [inline xml/System.Threading.Tasks/Task.xml:5453]:
```suggestion
An exception was thrown during
```

File: xml/System.Threading.Tasks/Task.xml [hunk 15]
```diff
@@ -5436,9 +5442,16 @@ timeout is greater than
         </Parameter>
       </Parameters>
       <Docs>
-        <param name="tasks">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="tasks">An array of <see cref="T:System.Threading.Tasks.Task" /> instances on which to wait.</param>
+        <summary>Waits for all of the provided <see cref="T:System.Threading.Tasks.Task" /> objects to complete execution.</summary>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">The <paramref name="tasks" /> argument contains a <see langword="null" /> element.</exception>
+        <exception cref="T:System.AggregateException">At least one of the <see cref="T:System.Threading.Tasks.Task" /> instances was canceled
```
Comment (gewarren) [inline xml/System.Threading.Tasks/Task.xml:5449]:
```suggestion
        <exception cref="T:System.AggregateException">At least one of the <see cref="T:System.Threading.Tasks.Task" /> instances was canceled.
```

File: xml/System.Threading.Tasks/Task.xml [hunk 16]
```diff
@@ -6463,10 +6482,20 @@ timeout is greater than
...
+          <para> If any of the supplied tasks completes in a faulted state, the returned task will also complete in a Faulted state, where its exceptions will contain the aggregation of the set of unwrapped exceptions from each of the supplied tasks.
+ </para>
+          <para> If none of the supplied tasks faulted but at least one of them was canceled, the returned task will end in the Canceled state.
+ </para>
+          <para> If none of the tasks faulted and none of the tasks were canceled, the resulting task will end in the RanToCompletion state.
+ </para>
+          <para> If the supplied span contains no tasks, the returned task will immediately transition to a RanToCompletion state before it's returned to the caller.
+ </para>
+        </remarks>
+        <exception cref="T:System.ArgumentException">The <paramref name="tasks" /> array contained a <see langword="null" /> task.</exception>
```
Comment (gewarren) [inline xml/System.Threading.Tasks/Task.xml:6498]:
```suggestion
        <exception cref="T:System.ArgumentException">The <paramref name="tasks" /> array contains a <see langword="null" /> task.</exception>
```

File: xml/System.Threading.Tasks/Task.xml [hunk 17]
```diff
@@ -6692,10 +6721,21 @@ timeout is greater than
...
+ </para>
+          <para> If none of the supplied tasks faulted but at least one of them was canceled, the returned task will end in the Canceled state.
+ </para>
+          <para> If none of the tasks faulted and none of the tasks were canceled, the resulting task will end in the RanToCompletion state.
+ The Result of the returned task will be set to an array containing all of the results of the supplied tasks in the same order as they were provided (e.g. if the input tasks array contained t1, t2, t3, the output task's Result will return an TResult[] where arr[0] == t1.Result, arr[1] == t2.Result, and arr[2] == t3.Result).
+ </para>
+          <para> If the supplied array/enumerable contains no tasks, the returned task will immediately transition to a RanToCompletion state before it's returned to the caller.  The returned TResult[] will be an array of 0 elements.
+ </para>
+        </remarks>
+        <exception cref="T:System.ArgumentException">The <paramref name="tasks" /> array contained a <see langword="null" /> task.</exception>
```
Comment (gewarren) [inline xml/System.Threading.Tasks/Task.xml:6738]:
```suggestion
        <exception cref="T:System.ArgumentException">The <paramref name="tasks" /> array contains a <see langword="null" /> task.</exception>
```

File: xml/System.Threading.Tasks/Task.xml [hunk 18]
```diff
@@ -6898,10 +6938,11 @@ timeout is greater than
...
       <Docs>
-        <param name="tasks">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="tasks">The tasks to wait on for completion.</param>
+        <summary>Creates a task that will complete when any of the supplied tasks have completed.</summary>
+        <returns>A task that represents the completion of one of the supplied tasks.  The return Task's Result is the task that completed.</returns>
+        <remarks>The returned task will complete when any of the supplied tasks has completed.  The returned task will always end in the RanToCompletion state with its Result set to the first task to complete.  This is <see langword="true" /> even if the first task to complete ended in the Canceled or Faulted state.</remarks>
+        <exception cref="T:System.ArgumentException">The <paramref name="tasks" /> array contained a <see langword="null" /> task, or was empty.</exception>
```
Comment (gewarren) [inline xml/System.Threading.Tasks/Task.xml:6945]:
```suggestion
        <exception cref="T:System.ArgumentException">The <paramref name="tasks" /> array contains a <see langword="null" /> task, or is empty.</exception>
```

File: xml/System.Threading.Tasks/Task.xml [hunk 19]
```diff
@@ -7138,10 +7179,11 @@ The returned task will complete when any of the supplied tasks has completed.  T
...
         <typeparam name="TResult">To be added.</typeparam>
-        <param name="tasks">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="tasks">The tasks to wait on for completion.</param>
+        <summary>Creates a task that will complete when any of the supplied tasks have completed.</summary>
+        <returns>A task that represents the completion of one of the supplied tasks.  The return Task's Result is the task that completed.</returns>
+        <remarks>The returned task will complete when any of the supplied tasks has completed.  The returned task will always end in the RanToCompletion state with its Result set to the first task to complete.  This is <see langword="true" /> even if the first task to complete ended in the Canceled or Faulted state.</remarks>
+        <exception cref="T:System.ArgumentException">The <paramref name="tasks" /> array contained a <see langword="null" /> task, or was empty.</exception>
```
Comment (gewarren) [inline xml/System.Threading.Tasks/Task.xml:7186]:
```suggestion
        <exception cref="T:System.ArgumentException">The <paramref name="tasks" /> array contains a <see langword="null" /> task, or is empty.</exception>
```

**Review by gewarren — APPROVED** [review]

File: xml/System.Threading.Tasks/Task.xml (hunk 1 above)
Comment (carlossanlop) [reply xml/System.Threading.Tasks/Task.xml:6489]: No. I need to fix it in the tool.

File: xml/System.Threading.Tasks/Task.xml [hunk 20]
```diff
@@ -6463,10 +6482,20 @@ timeout is greater than
...
+        <returns>A task that represents the completion of all of the supplied tasks.</returns>
+        <remarks>
+          <para> If any of the supplied tasks completes in a faulted state, the returned task will also complete in a Faulted state, where its exceptions will contain the aggregation of the set of unwrapped exceptions from each of the supplied tasks.
+ </para>
+          <para> If none of the supplied tasks faulted but at least one of them was canceled, the returned task will end in the Canceled state.
+ </para>
+          <para> If none of the tasks faulted and none of the tasks were canceled, the resulting task will end in the RanToCompletion state.
+ </para>
+          <para> If the supplied span contains no tasks, the returned task will immediately transition to a RanToCompletion state before it's returned to the caller.
+ </para>
```
Comment (carlossanlop) [inline xml/System.Threading.Tasks/Task.xml:6496]:
```suggestion
          <para>If any of the supplied tasks completes in a faulted state, the returned task will also complete in a Faulted state, where its exceptions will contain the aggregation of the set of unwrapped exceptions from each of the supplied tasks.
 </para>
          <para>If none of the supplied tasks faulted but at least one of them was canceled, the returned task will end in the Canceled state.
 </para>
          <para>If none of the tasks faulted and none of the tasks were canceled, the resulting task will end in the RanToCompletion state.
 </para>
          <para>If the supplied span contains no tasks, the returned task will immediately transition to a RanToCompletion state before it's returned to the caller.
 </para>
```

**Pull request merged**